src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -121,7 +121,7 @@ @implementation MTRDeviceController {
     MTRP256KeypairBridge _operationalKeypairBridge;
     MTRDeviceAttestationDelegateBridge * _deviceAttestationDelegateBridge;
     MTRDeviceControllerFactory * _factory;
-    NSMutableDictionary * _nodeIDToDeviceMap;
+    NSMapTable * _nodeIDToDeviceMap;
```

Comment:
Won't this potentially break some API consumers that get a device, set a delegate on it, then don't worry about the device anymore?   I guess at this point our API surface perhaps doesn't lead one to think this should work (though it does)?

Reply:
I'm worried about that too, however we need to not own these thing, as they need to clean up automatically. Usually this is done with ref counting, or an NSMapTable.

Reply:
Been talking with the home team about this, as they need to hold on to a reference